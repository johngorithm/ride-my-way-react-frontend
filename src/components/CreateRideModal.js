import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';

import InputGroup from 'components/Atom/InputGroup';
import Button from 'components/Atom/Button';

import { closeModal } from 'actions/modalActions';
import { createRide } from 'actions/rideActions';
import validateRide from 'utils/validateRide'

export class CreateRideModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      rideInfo: {
        destination: '',
        date: '',
        time: '',
        takeOffVenue: '',
        capacity: ''
      },
      errors: {
        destination: '',
        date: '',
        time: '',
        takeOffVenue: '',
        capacity: ''
      }
    };
  }

  handleChange = (e) => {
    const { rideInfo } = this.state;
    const newRideData = { ...rideInfo };

    newRideData[`${e.target.name}`] = e.target.value;

    this.setState({
      rideInfo: newRideData
    });
  }

  closeModal(e) {
    e.preventDefault();
    this.props.closeModal(this.props.modalName)
  }

  handleSubmit = event => {
    event.preventDefault();
    const { rideInfo } = this.state;

    const { errors, isValidData } = validateRide(rideInfo);

    if (isValidData) {
      this.setState({
        errors: {
          destination: '',
          date: '',
          time: '',
          takeOffVenue: '',
          capacity: ''
        }
      })
      return this.props.createRide(rideInfo).then(() => {
        this.props.closeModal(this.props.modalName)
        this.props.history.push('/home');
      })
    } else {
      this.setState({
        errors
      });
    }
  }

  render() {
    const { destination, date, time, capacity, takeOffVenue } = this.state.rideInfo;
    const { errors } = this.state;
    const { succesMessge, isLoading } = this.props;
    return (
      <Fragment>
        <div className="modal" id="add-offer-modal">
          <div className="modal-content animate">
            <form id="create-offer-form" onSubmit={this.handleSubmit}>
              <div id="form-header" className="co-xl-12 co-lg-12 co-md-12 co-sm-12" >
                  <h3 id="form-header-title" className ="center-text">OFFER A RIDE</h3>
                  {succesMessge ? (<p className="success-message small center-text">{succesMessge}</p>) : null}
                  
              </div>
              <div className="row">
                <InputGroup  
                  classes="co-xl-12 co-lg-12 co-md-12 co-sm-12"
                  label="Destination"
                  error={errors.destination}
                  type="text"
                  name="destination"
                  placeholder="where are you goin?"
                  value={destination}
                  onChange={this.handleChange}
                />

                <InputGroup  
                  classes="co-xl-6 co-lg-6 co-md-6 co-sm-6"
                  label="Date"
                  error={errors.date}
                  type="date"
                  name="date"
                  value={date}
                  onChange={this.handleChange}
                />

                <InputGroup  
                  classes="co-xl-6 co-lg-6 co-md-6 co-sm-6"
                  label="Time"
                  error={errors.time}
                  type="time"
                  name="time"
                  value={time}
                  onChange={this.handleChange}
                />

                <InputGroup  
                  classes="co-xl-12 co-lg-12 co-md-12 co-sm-12"
                  label="Capacity"
                  error={errors.capacity}
                  type="number"
                  name="capacity"
                  max="100"
                  placeholder="max: 100"
                  value={capacity}
                  onChange={this.handleChange}
                />

                <InputGroup  
                  classes="co-xl-12 co-lg-12 co-md-12 co-sm-12"
                  label="Takeoff Venue"
                  error={errors.takeOffVenue}
                  type="text"
                  name="takeOffVenue"
                  placeholder="where will your journey start?"
                  value={takeOffVenue}
                  onChange={this.handleChange}
                />

                <div className="co-xl-12 co-lg-12 co-md-12 co-sm-12 right-text">
                  <Button 
                    text="CLOSE" 
                    type="button" 
                    onClick={this.closeModal.bind(this)}
                    classes="button button-white close"
                  />
                  {" "}

                  <Button
                    text="CREATE OFFER"
                    type="submit"
                    style={ isLoading ? {opacity: 0.4, cursor: 'forbidden'} : null}
                    classes="button button-blue create" 
                    disabled={ isLoading ? true : false}
                  />
                </div>
              </div>
            </form>
          </div>
        </div>
      </Fragment>
    );
  }
}

CreateRideModal.defaultProps = {
  modalName: 'CreateRideModal'
}

CreateRideModal.propTypes = {
  isCreateRideModalOpen: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired,
  createRide: PropTypes.func.isRequired,
  modalName: PropTypes.string.isRequired,
  succesMessge: PropTypes.string,
  history: PropTypes.objectOf(PropTypes.any),
  isLoading: PropTypes.bool.isRequired
}


const mapStateToProps = (state) => ({
  isCreateRideModalOpen: state.modal.isCreateRideModalOpen,
  isLoading: state.ride.isLoading,
  succesMessge: state.ride.successMessge
});

export default withRouter(connect(
  mapStateToProps,
  { closeModal, createRide }
)(CreateRideModal));
