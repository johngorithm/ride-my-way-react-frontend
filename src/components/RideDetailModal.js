import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { closeModal } from 'actions/modalActions';

class RideDetailModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Fragment>
        <div className="modal" id="detail-modal">
          <div className="modal-content animate">
            <div className="tile">
              <div className="tile-heading center-text">
                <h4>
                  RIDE TO <span>{this.props.ride.destination}</span>
                </h4>
                <p className="smaller">
                  <strong>from: </strong>
                  <span>{this.props.ride.takeOffVenue}</span>
                </p>
                <span className="message smaller" />
              </div>

              <div className="tile-body">
                <div className="row">
                  <div className="co-xl-6 co-lg-6 co-md-6 co-sm-6 co-xs-6 center-text border">
                    <p className="data-heading small">Date</p>
                    <p className="small date">{this.props.ride.date}</p>
                  </div>
                  <div className="co-xl-6 co-lg-6 co-md-6 co-sm-6 co-xs-6 center-text">
                    <p className="data-heading small">Time</p>
                    <p className="small time">{this.props.ride.time}</p>
                  </div>
                </div>
              </div>

              <div id="capacity-info" className="tile-body not-first center-text">
                <p className="left">
                  capacity<span className="badge" id="capacity">{this.props.ride.capacity}</span>

                </p>
                <p className="right">
                  occupied<span className="badge" id="space-occupied"> {this.props.ride.occupied}</span>
                </p>
              </div>

              <div className="tile-body driver center-text padded">
                <p className="data-heading small">Driver</p>
                <p className="small driver">{this.props.ride.driver}</p>
              </div>

              <div className="tile-footer center-text">
                <button onClick={() => this.props.closeModal('RideDetailModal')} className="button button-blue close">CLOSE</button>
                <button
                  //onClick="joinRide(this)"
                  className="button button-white join"
                >
                  JOIN
                </button>
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

RideDetailModal.propTypes = {
  closeModal: PropTypes.func.isRequired,
  ride: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
  ride: state.ride.ride
})

export default connect(
  mapStateToProps,
  { closeModal }
)(RideDetailModal);
