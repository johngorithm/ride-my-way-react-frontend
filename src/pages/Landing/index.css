.container,
#root,
.App {
  height: 100%;
}

#landing-page {
  background-image: url('../../static/images/carpooling-apps-like-lyft-624x351.jpeg');
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  height: 100%;
}

#landing-page.content .wrapper {
  color: white;
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
  right: 0;
  left: 0;
}

#landing-page.content .wrapper p {
  color: white;
  font-size: 20px;
}

#landing-page.content .wrapper h2 {
  color: white;
  font-size: 50px;
  font-family: 'Montserrat', sans-serif;
  margin-bottom: 30px;
}

#landing-page.content .overlay {
  background: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

#landing-page.content .wrapper a .button {
  padding: 10px 40px;
  font-weight: bold;
}
