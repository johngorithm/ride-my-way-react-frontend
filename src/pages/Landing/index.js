import React from 'react';
import { Link } from 'react-router-dom';
import './index.css';

const Landing = () => {
  return (
    <main id="landing-page" className="dropdown content">
      <div className="overlay" />
      <div className="wrapper">
        <h2 className="welcome-title">Welcome</h2>
        <p className="welcome-message">
          Ride My Way is an app that lets you find a ride going your way or
          Create one and get others to their destination
        </p>

        <Link to="/login">
          <button className="button button-blue">GET STARTED</button>
        </Link>
      </div>
    </main>
  );
};

export default Landing;
