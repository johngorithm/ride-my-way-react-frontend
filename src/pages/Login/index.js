import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'


import { loginUser, deleteAuthErrorMessage } from 'actions/authActions';

import './login.css';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userInfo: {
        username: '',
        password: '',
      },
      errors: {}
    }
  }

  handleChange(e) {
    const user = { ...this.state.userInfo };
    user[`${e.target.name}`] = e.target.value;
    this.setState({
      userInfo: user
    });
  }

  handleSubmit(e) {
    e.preventDefault();

    // CLEAR ERROR MESSAGE
    this.props.deleteAuthErrorMessage();


    const errors = {};
    let isValidData = true;

    Object.entries(this.state.userInfo).forEach((field) => {
      const [fieldName, fieldData] = field;

      if (fieldData.trim() === '') {
        errors[fieldName] = `${fieldName} is required`;
        isValidData = false;
      } else if (fieldName === 'password' && fieldData.length < 6) {
        errors[fieldName] = 'password must NOT be less than 6 characters';
        isValidData = false;
      }
    });

    if (isValidData) {
      this.setState({
        errors: {}
      })
      // Login Action
      this.props.loginUser(this.state.userInfo);

    } else {
      this.setState({
        errors: errors
      })
    }
  }


  render() {
    return (
      <Fragment>
        <main id="login-page" className="content">
          <div className="overlay" />
          <div className="wrapper">
            <h3>LOGIN</h3>
            <form onSubmit={this.handleSubmit.bind(this)} >
              <p className="error-message" />
              <p className="success-message" />
              <input
                type="text"
                placeholder="Enter Username"
                name="username"
                required
                onChange={this.handleChange.bind(this)}
              />{this.state.errors.username ? <span className="error-message">{this.state.errors.username}</span> : null}
              <br />
              <input
                type="password"
                placeholder="Enter Password"
                name="password"
                required
                onChange={this.handleChange.bind(this)}
              />{this.state.errors.password ? <span className="error-message">{this.state.errors.password}</span> : null}
              <br />

              <button id="login-btn" type="submit" className="button button-blue">
                LOGIN
              </button>
            </form>
          </div>
        </main>
      </Fragment>
    );
  }
}

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  deleteAuthErrorMessage: PropTypes.func,
  history: PropTypes.any.isRequired
}

export default connect(
  null,
  { loginUser, deleteAuthErrorMessage }
)(Login);
