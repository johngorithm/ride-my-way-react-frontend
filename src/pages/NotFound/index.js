import React from 'react';
import './notfound.css';

const NotFound = () => {
  return (
    <div id="not-found">
      <h1>404 <br/> Page Not Found</h1>
    </div>
  );
};

export default NotFound;
