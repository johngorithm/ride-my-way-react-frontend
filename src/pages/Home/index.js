import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'

import { openModal } from 'actions/modalActions';
import { getRides } from 'actions/rideActions';
import { checkSession } from 'actions/authActions';
import SubMenu from 'components/SubMenu';
import CreateRideModal from 'components/CreateRideModal';
import RideDetailModal from 'components/RideDetailModal';
import RideCard from 'components/RideCard';

import './home.css';

class Home extends React.Component {
  constructor(props) {
    super(props);
  }

  componentWillMount() {
    this.props.checkSession().then(() => {
      if (!this.props.isAuthenticated) {
        return this.props.history.push('/login');
      } else {
        this.props.getRides();
      }
    });
  }

  checkOwnership = (ride, user) => {
    let ownership = '';
    if (ride.creator_id === user.user_id ) {
      return ownership = 'yours';
    }
    return ownership
  }

  render() {
    const { rides } = this.props;

    return (
      <Fragment>
        {this.props.isCreateRideModalOpen ? <CreateRideModal /> : null}
        {this.props.isRideDetailModalOpen ? <RideDetailModal /> : null}

        <SubMenu />

        <main className="wrapper dropdown">
          <div className="row" id="rides-loader">
            {
              rides ? rides.map(ride => {
                const user = JSON.parse(localStorage.getItem('rmwUser'));
                const ownership = this.checkOwnership(ride, user);
                return <RideCard key={ride.ride_id} ride={ride} ownership={ownership} />
              }) : (<div id="loading" className="center-text">
              <p>LOADING ...</p>
            </div>)
            }

          </div>
        </main>
      </Fragment>
    )
  }
}

Home.propTypes = {
  isCreateRideModalOpen: PropTypes.bool.isRequired,
  isRideDetailModalOpen: PropTypes.bool.isRequired,
  openModal: PropTypes.func.isRequired,
  rides: PropTypes.array.isRequired,
  getRides: PropTypes.func.isRequired,
  history: PropTypes.any,
  isAuthenticated: PropTypes.bool.isRequired,
  checkSession: PropTypes.func.isRequired
}


const mapStateToProps = (state) => ({
  isCreateRideModalOpen: state.modal.isCreateRideModalOpen,
  isRideDetailModalOpen: state.modal.isRideDetailModalOpen,
  rides: state.ride.rides,
  isAuthenticated: state.auth.isAuthenticated
})



export default connect(
  mapStateToProps,
  { openModal, getRides, checkSession }
)(Home);
