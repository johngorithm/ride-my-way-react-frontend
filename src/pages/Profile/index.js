import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'


import CreateRideModal from '../../components/CreateRideModal';
import RideDetailModal from '../../components/RideDetailModal';
import './profile.css';

class Profile extends React.Component {
  render() {
    return (
      <Fragment>
        {this.props.isCreateRideModalOpen ? <CreateRideModal /> : null}
        {this.props.isRideDetailModalOpen ? <RideDetailModal /> : null}

        <main id="profile-loader">
          <p id="general-message" />

          <header id="profile-header">
            <div id="user-info" className="row">
              <p id="user-loader">Loading ...</p>
            </div>
          </header>

          <section id="ride-details">
            <div className="row wrapper">
              <div className="co-xl-6 co-lg-6 co-md-6 co-sm-12">
                <div className="square">
                  <p>RIDES OFFERED</p>
                  <h1 id="offered">Loading...</h1>
                </div>
              </div>

              <div className="co-xl-6 co-lg-6 co-md-6 co-sm-12">
                <div className="square">
                  <p>RIDES TAKEN</p>
                  <h1 id="taken">Loading...</h1>
                </div>
              </div>
            </div>
          </section>

          <section id="rides-offered" className="hori-display">
            <div className="wrapper">
              <h5 className="section-intro center-text">RIDES OFFERED</h5>
              <div className="container center-text">
                <p id="rides-offered-loader " className="loading">
                  Loading ...
                </p>
              </div>
            </div>
          </section>

          <section id="rides-taken" className="hori-display">
            <div className="wrapper">
              <h5 className="section-intro center-text">RIDES TAKEN</h5>
              <div className="container center-text">
                <p id="rides-taken-loader" className="loading">
                  Loading ...
                </p>
              </div>
            </div>
          </section>
        </main>
      </Fragment>
    );
  }
}

Profile.propTypes = {
  isCreateRideModalOpen: PropTypes.bool.isRequired,
  isRideDetailModalOpen: PropTypes.bool.isRequired
}

const mapStateToProps = (state) => ({
  isCreateRideModalOpen: state.modal.isCreateRideModalOpen,
  isRideDetailModalOpen: state.modal.isRideDetailModalOpen
});

export default connect(
  mapStateToProps
)(Profile);
