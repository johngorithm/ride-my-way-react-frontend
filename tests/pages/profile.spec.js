import React from 'react';
import { shallow } from 'enzyme';
import {Profile} from 'pages/Profile';

import CreateRideModal from 'components/CreateRideModal';
import RideDetailModal from 'components/RideDetailModal';


const props = {
  isCreateRideModalOpen: true,
  isRideDetailModalOpen: true
}
describe('USER PROFILE TEST', () => {
  const wrapper = shallow(<Profile {...props} />);
  
  test('renders successfully', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('does not rendr modals when not require', () => {
    wrapper.setProps({
      isCreateRideModalOpen: false,
      isRideDetailModalOpen: false
    });

    expect(wrapper.find(<CreateRideModal />).exists()).toBe(false);
    expect(wrapper.find(<RideDetailModal />).exists()).toBe(false);
  })
});