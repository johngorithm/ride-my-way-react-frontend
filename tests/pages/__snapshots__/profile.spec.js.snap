// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USER PROFILE TEST matches snapshot 1`] = `
<Fragment>
  <withRouter(Connect(CreateRideModal)) />
  <Connect(RideDetailModal) />
  <main
    id="profile-loader"
  >
    <p
      id="general-message"
    />
    <header
      id="profile-header"
    >
      <div
        className="row"
        id="user-info"
      >
        <p
          id="user-loader"
        >
          Loading ...
        </p>
      </div>
    </header>
    <section
      id="ride-details"
    >
      <div
        className="row wrapper"
      >
        <div
          className="co-xl-6 co-lg-6 co-md-6 co-sm-12"
        >
          <div
            className="square"
          >
            <p>
              RIDES OFFERED
            </p>
            <h1
              id="offered"
            >
              Loading...
            </h1>
          </div>
        </div>
        <div
          className="co-xl-6 co-lg-6 co-md-6 co-sm-12"
        >
          <div
            className="square"
          >
            <p>
              RIDES TAKEN
            </p>
            <h1
              id="taken"
            >
              Loading...
            </h1>
          </div>
        </div>
      </div>
    </section>
    <section
      className="hori-display"
      id="rides-offered"
    >
      <div
        className="wrapper"
      >
        <h5
          className="section-intro center-text"
        >
          RIDES OFFERED
        </h5>
        <div
          className="container center-text"
        >
          <p
            className="loading"
            id="rides-offered-loader "
          >
            Loading ...
          </p>
        </div>
      </div>
    </section>
    <section
      className="hori-display"
      id="rides-taken"
    >
      <div
        className="wrapper"
      >
        <h5
          className="section-intro center-text"
        >
          RIDES TAKEN
        </h5>
        <div
          className="container center-text"
        >
          <p
            className="loading"
            id="rides-taken-loader"
          >
            Loading ...
          </p>
        </div>
      </div>
    </section>
  </main>
</Fragment>
`;
